dash/include/dash/Coarray.h

```diff
@@ -0,0 +1,771 @@
+   */
+  template<
+    class MEMTYPE,
+    class P=T,
+    int __rank = _rank::value,
+    typename = typename std::enable_if<(__rank == 0)>::type>
+  inline MEMTYPE & member(const MEMTYPE P::*mem) {
+    size_t offs = (size_t) &( reinterpret_cast<P*>(0)->*mem);
+    return member<MEMTYPE>(offs);
+  }
+
+  /**
+   * allows fortran like local access of scalars
+   * \code
+   *   dash::Coarray<int> i;
+   *   i = 42;
+   *   i += 21;
+   * \endcode
+   */
+  template<
+    int __rank = _rank::value,
+    typename = typename std::enable_if<(__rank == 0)>::type>
+  inline value_type & operator +=(const value_type & value) {
+    return *(_storage.lbegin()) += value;
+  }
+  
+  /**
+   * allows fortran like local access of scalars
+   */
+  template<
+    int __rank = _rank::value,
+    typename = typename std::enable_if<(__rank == 0)>::type>
+  inline value_type & operator -=(const value_type & value) {
+    return *(_storage.lbegin()) -= value;
+  }
+  
+  /**
+   * allows fortran like local access of scalars
+   */
+  template<
+    int __rank = _rank::value,
+    typename = typename std::enable_if<(__rank == 0)>::type>
+  inline value_type & operator *=(const value_type & value) {
+    return *(_storage.lbegin()) *= value;
+  }
+  
+  /**
+   * allows fortran like local access of scalars
+   */
+  template<
+    int __rank = _rank::value,
+    typename = typename std::enable_if<(__rank == 0)>::type>
+  inline value_type & operator /=(const value_type & value) {
+    return *(_storage.lbegin()) /= value;
+  }
+  
+  /**
+   * allows fortran like local access of scalars
+   * \code
+   *   dash::Coarray<int> i;
+   *   i = 42;
+   *   int b = i + 21;
+   * \endcode
+   */
+  template<
+    int __rank = _rank::value,
+    typename = typename std::enable_if<(__rank == 0)>::type>
+  inline value_type operator +(const value_type & value) const {
+    return *(_storage.lbegin()) + value;
+  }
+  
+  /**
+   * allows fortran like local access of scalars
+   */
+  template<
+    int __rank = _rank::value,
+    typename = typename std::enable_if<(__rank == 0)>::type>
+  inline value_type operator -(const value_type & value) const {
+    return *(_storage.lbegin()) - value;
+  }
+  
+  /**
+   * allows fortran like local access of scalars
+   */
+  template<
+    int __rank = _rank::value,
+    typename = typename std::enable_if<(__rank == 0)>::type>
+  inline value_type operator *(const value_type & value) const {
+    return *(_storage.lbegin()) * value;
+  }
+  
+  /**
+   * allows fortran like local access of scalars
+   */
+  template<
+    int __rank = _rank::value,
+    typename = typename std::enable_if<(__rank == 0)>::type>
+  inline value_type operator /(const value_type & value) const {
+    return *(_storage.lbegin()) / value;
+  }
+  
+  /**
+   * allows fortran like local access of scalars
+   */
+  template<
+    int __rank = _rank::value,
+    typename = typename std::enable_if<(__rank == 0)>::type>
+  inline value_type operator ++() {
```

Comment:
Same as above.

Reply:
Actually, `operator ++` should return `value_type &` then, otherwise something like `++(++coarr[i])` won't work correctly.

Also, should these operators be disabled/altered for atomics?